src/main/java/uk/gov/hmcts/reform/roleassignmentrefresh/domain/service/process/RefreshJobsOrchestrator.java

```diff
@@ -109,11 +147,141 @@ private void sendJobToORMService(Long jobId, UserRequest userRequest) {
+
+            if (jurisdictionMatch && roleCategoryMatch && roleNameMatch) {
+                returnCount = count;
+            }
+        }
+        return returnCount;
+    }
+
+    boolean isJurisdictionMatch(Count count, Count needle) {
+        if (count.getJurisdiction() == null && needle.getJurisdiction() == null) {
+            return true;
+        } else if (count.getJurisdiction() != null && needle.getJurisdiction() != null) {
+            return count.getJurisdiction().equals(needle.getJurisdiction());
+        } else {
+            return false;
+        }
+    }
+
+    boolean isRoleNameMatch(Count count, Count needle) {
+        if (count.getRoleName() == null && needle.getRoleName() == null) {
+            return true;
+        } else if (count.getRoleName() != null && needle.getRoleName() != null) {
+            return count.getRoleName().equals(needle.getRoleName());
+        } else {
+            return false;
+        }
+    }
+
+    public void updateWithAfterCountOrCreateNew(Count afterCount, List<Count> counts) {
+        Count matchedCount = findMatchingCount(afterCount, counts);
+        if (matchedCount != null) {
+            matchedCount.updateAfterCount(afterCount.getAfterCount());
+        } else {
+            afterCount.setDifference(afterCount.getAfterCount());
+            counts.add(afterCount);
+        }
     }
 
+    public void sendEmailWithCounts(CountResponse responseBeforeRefresh, CountResponse responseAfterRefresh,
+                                    List<RefreshJob> refreshJobs) {
+        List<Count> jurisdictionCount = compareCounts(responseBeforeRefresh, responseAfterRefresh,
+                ORG_USER_COUNT_BY_JURISDICTION);
+        List<Count> jurisdictionAndRoleNameCount = compareCounts(responseBeforeRefresh, responseAfterRefresh,
+                ORG_USER_COUNT_BY_JURISDICTION_AND_ROLE_NAME);
+
+        Map<String, Object> templateMap = new HashMap<>();
+
+        templateMap.put("refreshJobs", refreshJobs);
+        templateMap.put("jurisdictionCount", jurisdictionCount);
+        templateMap.put("jurisdictionAndRoleNameCount", jurisdictionAndRoleNameCount);
+
+        EmailData emailData = EmailData
+                .builder()
+                .emailSubject(EMAIL_SUBJECT)
+                .templateMap(templateMap)
+                .build();
+        emailService.sendEmail(emailData);
+    }
+
+    @JsonPropertyOrder({ "jurisdiction", "roleCategory", "count"})
+    public interface JurisdictionRoleCategoryAndCount {
+        String getJurisdiction();
+
+        String getRoleCategory();
+
+        BigInteger getCount();
+    }
+
+    @JsonPropertyOrder({ "jurisdiction", "roleCategory", "roleName", "count"})
+    public interface JurisdictionRoleCategoryNameAndCount {
+        String getJurisdiction();
+
+        String getRoleCategory();
+
+        String getRoleName();
+
+        BigInteger getCount();
+    }
```

Comment:
Can you remove these interaces if they are no longer used. i.e. I assume we are using the new model `CountResponse.CountData`.

Reply:
removed

Reply:
Resolved in a3d32427ca1ed13333909504af43ca392313e622.